AU2/plugins/custom_plugins/PolicePlugin.py

```diff
@@ -88,122 +93,205 @@ def __init__(self):
                 "Police -> Set Ranks",
                 self.ask_set_ranks,
                 self.answer_set_ranks
+            ),
+            ConfigExport(
+                "PolicePlugin_toggle_auto",
+                "Police -> Toggle Auto Ranking",
+                lambda *args: [],
+                self.answer_toggle_auto
+            ),
+            ConfigExport(
+                "PolicePlugin_toggle_manual",
+                "Police -> Toggle Manual Ranking",
+                lambda *args: [],
+                self.answer_toggle_manual
+            ),
+            ConfigExport(
+                "PolicePlugin_toggle_police_kills_rankup",
+                "Police -> Toggle Ranking up on Police Kills",
+                lambda *args: [],
+                self.answer_toggle_police_kills_rankup
             )
         ]
 
-        self.ranks = GENERIC_STATE_DATABASE.arb_state.setdefault(self.identifier, DEFAULT_RANKS)
+    # I've done the database stuff in such a way that configs are only stored if changed; otherwise the hardcoded default configs are used.
+    # Not sure how sensible this is, but it seems to work
+
+    def gsdb_get(self, plugin_state_id):
+        return GENERIC_STATE_DATABASE.arb_state.get(self.identifier, {}).get(self.plugin_state[plugin_state_id]['id'], self.plugin_state[plugin_state_id]['default'])
```

Comment:
nit: Can you try line-wrapping this (you can end a line with a backslash to avoid breaking it, e.g. something like this (but don't commit exactly this suggestion without testing)

```suggestion
    def gsdb_get(self, plugin_state_id):
        return GENERIC_STATE_DATABASE.arb_state   \
                        .get(self.identifier, {}) \
                        .get(self.plugin_state[plugin_state_id]['id'], self.plugin_state[plugin_state_id]['default'])
```